### PR TITLE
Information on new `ref` syntax for by-reference cells

### LINKED ISSUE #1046: Update "passing memories by reference" doc to use `ref cell` syntax
With @paili0628's work on `ref cell` syntax, we don't have to use the hacky way to pass memories by reference.

### PR DESCRIPTION
**Pull request by rachitnigam**
Fixes #1046

### CODE REVIEW

File: docs/lang/memories-by-reference.md [hunk 1]
```diff
@@ -15,13 +15,58 @@ int main() {
...
+The language provides two ways to doing this.
+
+## The Easy Way
+
+Calyx uses the `ref` keyword to describe cells that are passed in by-reference:
+
+```
+component add_one() -> () {
+  cells {
+    ref mem = std_mem_d1(32, 4, 3); // A memory passed in by refernece.
```
Comment (sampsyo) [inline docs/lang/memories-by-reference.md:33]: refernece -> reference

File: docs/lang/memories-by-reference.md [hunk 2]
```diff
@@ -15,13 +15,58 @@ int main() {
...
+
+This component define `mem` as a memory that is passed in by reference to the component.
+Inside the component we can use the cell like any other cell in the program.
+
+Next, to pass the memory to the component, we can use the `invoke` syntax:
+```
+component add_one() -> () { ... }
+component main() -> () {
+  cells {
+    A = std_mem_d1(32, 4, 3); // A memory passed in by refernece.
```
Comment (sampsyo) [inline docs/lang/memories-by-reference.md:48]: The comment does not apply here. Maybe we should point out that this is an ordinary owned subcomponent instead?

**Review by sampsyo — COMMENTED** [review]
This is really helpful; thanks!! I was just going to suggest that we write some docs for this, because it is suddenly such a core component of the language.

What do you think about changing the title/intro here to be about passing in _arbitrary components_ by reference, and just to use `std_mem_d1` as one component you might want to share in this way? If you approve, @rachitnigam and/or @paili0628, I can take the baton on that.

**Comment by rachitnigam** [comment]
@sampsyo that sounds good! The system does allow for passing arbitrary cells by reference so we should update it. Please free to make the edits you proposed.

**Pull request merged**